modopt/opt/proximity/denoiser.py

```diff
@@ -0,0 +1,129 @@
+"""Denoiser Operators.
+
+Denoising operators that can be used as a drop-in replacement of proximal
+operators in iterative algorithms, making them a "plug-and-play" method.
+
+TODO Add support for VBM4D (video denoising, may be usefull for dynamic data).
+TODO Add support for Local Low-Rank Denoising.
+"""
+
+from modopt.opt.proximity.base import ProximityParent
+from modopt.signal.local_pca import local_svd_thresh
+
+BM3D_AVAILABLE = True
+try:
+    import bm3d
+except ImportError:
+    BM3D_AVAILABLE = False
+
+
+class BM3Denoise(ProximityParent):
+    """
+    BM3D Denoiser.
+
+    This a wrapper around the BM3D Denoiser, using the implementation of
+    :cite:`makinen2020`.
+    """
+
+    def __init__(self, noise_std=1.0):
+        self.noise_std = noise_std
+        self.op = self._op_method
+        self.cost = lambda x_val: 0
+
+    def _op_method(self, input_data, extra_factor=1.0):
+        return bm3d.bm3d(
+            input_data,
+            sigma_psd=extra_factor * self.noise_std,
+            sigma_arg=bm3d.BM3DStages.ALL_STAGES,
+        )
+
+
+class LocalLowRankDenoiser(ProximityParent):
+    """Local Low Rank Denoiser.
+
+    This a python implementation of
+    "NOise Reduction with DIstribution Corrected" (NORDIC).
+
+    It follows the principles described in :cite:`moeller2021` and
+    the MATLAB implementation of  [1]
+
+
+    Parameters
+    ----------
+    patch_size: tuple
+        the patch size to consider for the local low rank regularisation
+    patch_overlap:
+        the amount of pixel/voxel overlap of each patch.
+        If > 0 the patch are averaged.
```

Comment:
Parameters section incomplete and it looks like `patch_size` should be `patch_shape`.